azure-storage-blob/azure/storage/blob/blob_service_client.py

```diff
@@ -88,13 +121,63 @@ def generate_shared_access_signature(
             ip=None,  # type: Optional[str]
             protocol=None  # type: Optional[str]
         ):
-        pass
+        '''
+        Generates a shared access signature for the blob service.
+        Use the returned signature with the sas_token parameter of any BlobService.
+        :param ResourceTypes resource_types:
+            Specifies the resource types that are accessible with the account SAS.
+        :param AccountPermissions permission:
```

Comment:
I had usually seen the `type` specified separately, which is kind of cumbersome. Do you have a link to any documentation for this format of `param` with the type and name together? I'm used to using Google style doctring syntax, and ironically actual documentation on this "low level" docstring format seems hard to come by.